mypy/partially_defined.py

```diff
@@ -396,3 +418,20 @@ def visit_with_stmt(self, o: WithStmt) -> None:
             expr.accept(self)
             self.process_lvalue(idx)
         o.body.accept(self)
+
+    def visit_import(self, o: Import) -> None:
+        for mod, alias in o.ids:
+            names = mod.split(".")
+            if alias is not None:
+                names[-1] = alias
+            for name in names:
+                self.tracker.record_definition(name)
```

Comment:
Actually I don't think this logic is correct, after `import x.y` only `x` becomes defined (try `import os.path; path` at runtime). While if you do `import x.y as z`, then only `z` becomes defined.

Reply:
Yep, you're right! I had no idea. Fixed